src/chem/aeronet_aod2ioda.py

```diff
@@ -104,76 +97,92 @@ def add_data(infile):
-    loc_mdata['surface_type'] = np.full((nlocs), 1)
-
-    c = np.empty([nlocs], dtype='S50')
-    c[:] = np.array(f3.siteid)
-    loc_mdata['station_id'] = writer.FillNcVector(c, 'string')
+    varAttrs = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
 
-    # Define datetime
-    d = np.empty([nlocs], 'S20')
-    for i in range(nlocs):
-        d[i] = f3.time[i].strftime('%Y-%m-%dT%H:%M:%SZ')
-    loc_mdata['datetime'] = writer.FillNcVector(d, 'datetime')
+    obsvars = {'aerosol_optical_depth': ['aod_340nm', 'aod_380nm',
+                                         'aod_440nm', 'aod_675nm',
+                                         'aod_500nm', 'aod_870nm',
+                                         'aod_1020nm', 'aod_1640nm']}
+
+    AttrData = {
+        'converter': os.path.basename(__file__),
+    }
+
+    DimDict = {
+    }
+
+    VarDims = {
+        'aerosol_optical_depth': ['nlocs', 'nchans'],
+        'frequency': ['nchans'],
+        'sensor_channel': ['nchans']
+    }
 
-    # Define var_mdata
-    print('Define var_mdata')
-    var_mdata['frequency'] = writer.FillNcVector(frequency, 'float')
-    var_mdata['sensor_channel'] = writer.FillNcVector(aod_chan, 'integer')
+    for key, value in obsvars.items():
+        varDict[key]['valKey'] = key, iconv.OvalName()
+        varAttrs[key, iconv.OvalName()]['_FillValue'] = -999.
+        varAttrs[key, iconv.OvalName()]['coordinates'] = 'longitude latitude station_elevation'
+        varAttrs[key, iconv.OvalName()]['units'] = 'unitless'
+        varDict[key]['errKey'] = key, iconv.OerrName()
+        varAttrs[key, iconv.OerrName()]['_FillValue'] = -999.
+        varAttrs[key, iconv.OerrName()]['units'] = 'unitless'
+        varAttrs[key, iconv.OerrName()]['coordinates'] = 'longitude latitude station_elevation'
+        varDict[key]['qcKey'] = key, iconv.OqcName()
+        varAttrs[key, iconv.OqcName()]['_FillValue'] = -999
+        varAttrs[key, iconv.OqcName()]['coordinates'] = 'longitude latitude station_elevation'
+        varAttrs[key, iconv.OqcName()]['units'] = ''
 
     for key, value in obsvars.items():
-        outdata[varDict[key]['valKey']] = np.array(f3[value].fillna(nc.default_fillvals['f4']))
-        outdata[varDict[key]['qcKey']] = np.where(outdata[varDict[key]['valKey']] == nc.default_fillvals['f4'],
+        outdata[varDict[key]['valKey']] = np.array(f3[value].fillna(np.float32(-999.)))
+        outdata[varDict[key]['qcKey']] = np.where(outdata[varDict[key]['valKey']] == np.float32(-999.),
                                                   1, 0)
-        outdata[varDict[key]['errKey']] = np.where(outdata[varDict[key]['valKey']] == nc.default_fillvals['f4'],
-                                                   nc.default_fillvals['f4'], 0.02)
+        outdata[varDict[key]['errKey']] = np.where(outdata[varDict[key]['valKey']] == np.float32(-999.),
+                                                   np.float32(-999.), 0.02)
```

Comment:
Example for writing single-precision float data (see comment above).

Reply:
@rhoneyager Thanks, Ryan. I have updated the bit type for AOD values. Not sure why the rest three checks still failed.